onnxruntime/core/session/provider_bridge_ort.cc

```diff
@@ -1955,6 +1955,47 @@ ORT_API_STATUS_IMPL(OrtApis::GetCUDAProviderOptionsAsString, _In_ const OrtCUDAP
   API_IMPL_END
 }
 
+ORT_API_STATUS_IMPL(OrtApis::UpdateCUDAProviderOptionsWithValue,
+                    _Inout_ OrtCUDAProviderOptionsV2* cuda_options,
+                    _In_ const char* key,
+                    _In_ void* value) {
+  API_IMPL_BEGIN
+#ifdef USE_CUDA
+  if (strcmp(key, "user_compute_stream") == 0) {
+    cuda_options->has_user_compute_stream = 1;
+    cuda_options->user_compute_stream = value;
```

Comment:
You could combine to two fields to just one: std::optional<CUDASteram*>.

Reply:
These two fields are defined in OrtCUDAProviderOptions (public C API) and OrtCUDAProviderOptionsV2. So it needs to change many code if we want to combine these two fields.